### PR TITLE
Removed rule to test only PostgresSQL

### PR DESCRIPTION
**Pull request by blafond**
enables testing during CI build of wrong credentials for non-PostgresSQL DB's to verify if the exception still exists or not for [#511](https://github.com/hibernate/hibernate-reactive/issues/511)

### CODE REVIEW

**Comment by DavideD** [comment]
@blafond I've changed this to a Draft PR.

Also, you can test this on your personal GitHub repository changing the condition here (if needed): https://github.com/hibernate/hibernate-reactive/blob/master/.github/workflows/build.yml#L4

**Comment by blafond** [comment]
Looks like MySQL DB test worked fine, but DB2 failed. Ran locally and the exception didn't contain `BogusBogus`

The exception was:
`io.vertx.db2client.DB2Exception: Invalid credentials, verify the user and password values supplied are correct`

So do we need to customize the test(s) to check to DB type? since the message from PgException won't be the same as DB2Exception?

**Comment by DavideD** [comment]
> sorry.. looks like MySQL errored too in the CI test

That was the whole point of the issue :-)
Could you checked why it has failed?

It's better if you enable this test only for Postgres and MySQL, so that error on Db2 won't appear

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/WrongCredentialsTest.java [hunk 1]
```diff
@@ -8,36 +8,35 @@
...
-import static org.hibernate.reactive.containers.DatabaseConfiguration.DBType.POSTGRESQL;
-
 /**
  * Check that the right exception is thrown when there is an error with the credentials.
  * <p>
  *     Similar to {@link org.hibernate.reactive.configuration.ReactiveConnectionPoolTest} but at the session level.
- *     Note that the wrong credentials are also used for the schema generation but exceptions are ignored during
- *     schema generation. You might just see some warnings in the log.
+ *     Note that the wrong configuration is also used for the schema generation but ORM catch the exception and
+ *     continue anyway it in case of errors. You might just see sine warnings in the log.
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/WrongCredentialsTest.java:24]: Wait, did you change the javadoc or just copied and pasted the one in the old test?

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/WrongCredentialsTest.java [hunk 2]
```diff
@@ -49,11 +48,22 @@ public void testWithTransaction(TestContext context) {
...
 				} ) );
 	}
 
+	private boolean isWrongCredentialsMessage(Throwable e ) {
+		// MySQL and PostgreSQL will contain the invalid credential value
+		// Cockroach will be lower case version of the wrong user
+		// Db2 will just state it
+		if( e.getMessage().toLowerCase().contains( BOGUS_USER.toLowerCase() ) ||
+				e.getMessage().contains( "Invalid credentials") ) {
+			return true;
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/WrongCredentialsTest.java:62]: Maybe just return the condition?
```
return e.getMessage().toLowerCase().contains( BOGUS_USER.toLowerCase() ) || e.getMessage().contains( "Invalid credentials")
```

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/WrongCredentialsTest.java (hunk 1 above)
Comment (blafond) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/WrongCredentialsTest.java]: copy paste... reverted text

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/WrongCredentialsTest.java (hunk 2 above)
Comment (blafond) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/WrongCredentialsTest.java:62]: sure.. less lines

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/WrongCredentialsTest.java [hunk 3]
```diff
@@ -27,17 +25,18 @@
  * </p>
  */
 public class WrongCredentialsTest extends BaseReactiveTest {
+	private static final CoreMessageLogger LOG = CoreLogging.messageLogger( WrongCredentialsTest.class );
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/WrongCredentialsTest.java:28]: Do you really need the LOG?

**Review by DavideD — CHANGES_REQUESTED** [review]
Looks good.
A couple of unnecessary lines of code

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/WrongCredentialsTest.java (hunk 3 above)
Comment (blafond) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/WrongCredentialsTest.java:28]: no

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/WrongCredentialsTest.java [hunk 4]
```diff
@@ -27,17 +25,18 @@
...
+
 
-	// I've decided to not test it for all databases because
-	// the error message might be different and it shouldn't be necessary
-	@Rule
-	public DatabaseSelectionRule selectionRule = DatabaseSelectionRule.runOnlyFor( POSTGRESQL );
+	private Configuration currentConfiguration;
```
[the remark this answers is not in the document]
Comment (blafond) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/WrongCredentialsTest.java:31]: already removed it

**Comment by gavinking** [comment]
thanks @blafond

**Pull request merged**